### PR TITLE
Add regression test for StringBuilder AppendLine

### PR DESCRIPTION
**Pull request by jnobile**
This test fails the [GreaterOrEqual](https://github.com/Faithlife/FaithlifeAnalyzers/blob/6a56cb914b3e4a1fb0f45b7e767e941bc8feee52/tests/Faithlife.Analyzers.Tests/Helpers/DiagnosticVerifier.Helper.cs#L64) assertion in `GetSortedDiagnosticsFromDocuments` with:

```
 ValidAppendLine Failed:   Argument 2 must be passed with the 'ref' keyword
Argument 2 must be passed with the 'ref' keyword
Expected: greater than or equal to 4
But was:  0

 at Faithlife.Analyzers.Tests.DiagnosticVerifier.GetSortedDiagnosticsFromDocuments(DiagnosticAnalyzer analyzer, Document[] documents) in C:\Code\FaithlifeAnalyzers\tests\Faithlife.Analyzers.Tests\Helpers\DiagnosticVerifier.Helper.cs:line 64
 at Faithlife.Analyzers.Tests.DiagnosticVerifier.GetSortedDiagnostics(String[] sources, String language, DiagnosticAnalyzer analyzer) in C:\Code\FaithlifeAnalyzers\tests\Faithlife.Analyzers.Tests\Helpers\DiagnosticVerifier.Helper.cs:line 39
 at Faithlife.Analyzers.Tests.DiagnosticVerifier.VerifyDiagnostics(String[] sources, String language, DiagnosticAnalyzer analyzer, DiagnosticResult[] expected) in C:\Code\FaithlifeAnalyzers\tests\Faithlife.Analyzers.Tests\Verifiers\DiagnosticVerifier.cs:line 87
 at Faithlife.Analyzers.Tests.DiagnosticVerifier.VerifyCSharpDiagnostic(String source, DiagnosticResult[] expected) in C:\Code\FaithlifeAnalyzers\tests\Faithlife.Analyzers.Tests\Verifiers\DiagnosticVerifier.cs:line 41
 at Faithlife.Analyzers.Tests.InterpolatedStringTests.ValidAppendLine() in C:\Code\FaithlifeAnalyzers\tests\Faithlife.Analyzers.Tests\InterpolatedStringTests.cs:line 58
```

### CODE REVIEW

**Comment by bgrainger** [comment]
Is this just a bug report (via adding a failing test) or is a fix still incoming (in which case this maybe should have been marked as a Draft PR)? I'm not sure what this PR is for.

**Comment by jnobile** [comment]
Now this is a bug report since it is failing; I was unsure if it was just my environment. It isn't a draft because I wanted the checks to run.

**Comment by ddunkin** [comment]
I think it's a mismatch between compiler and runtime. That overload was added in .NET 6. This test fails with a different error if `Microsoft.CodeAnalysis.CSharp.Workspaces` is updated.
`// Test0.cs(6,95): warning FL0014: Avoid using an interpolated string where an equivalent literal string exists.`

File: src/Faithlife.Analyzers/Faithlife.Analyzers.csproj [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.3" PrivateAssets="all" />
-    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.5.0" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.1.0" PrivateAssets="all" />
```
Comment (ddunkin) [inline src/Faithlife.Analyzers/Faithlife.Analyzers.csproj:14]: IfNotNull analzyer fails on anything newer than 3.5.0.

Comment (jnobile) [reply src/Faithlife.Analyzers/Faithlife.Analyzers.csproj]: Sure enough. 🤦 Wasn't expecting any yak shaving, to be honest.

**Pull request closed without merging**